strawberry/field.py

```diff
@@ -105,11 +122,51 @@ def __call__(self, resolver: _RESOLVER_TYPE) -> "StrawberryField":
 
     @property
     def arguments(self) -> List[StrawberryArgument]:
+        arguments_as_annotations = self.get_arguments()
+
         if not self.base_resolver:
             return []
 
+        # Convert the arguments to StrawberryArgument types
+        arguments = []
+        for arg_name, annotation in arguments_as_annotations.items():
+            if isinstance(annotation, BareStrawberryArgument):
+                argument = annotation
+            else:
+                default = self.base_resolver.get_argument_default(arg_name)
+                argument = self.create_argument(annotation, default)
+
+            arguments.append(argument.as_argument(arg_name))
+
+        return arguments
+
+    def get_arguments(self) -> Dict[str, object]:
+        """
+        Hook to modify the GraphQL arguments for the field
+        """
+        if not self.base_resolver:
+            return {}
+
         return self.base_resolver.arguments
 
+    def create_argument(
+        self, type_annotation: object, default: Any = UNSET
+    ) -> BareStrawberryArgument:
+        """
+        Helper function to create StrawberryArgument
+        """
```

Comment:
```suggestion
        """Helper function to create StrawberryArgument"""
```
(nitpicky suggestion for a draft PR, sorry 😅)